lib/dialects/mysql/query-generator.js

```diff
@@ -189,7 +189,8 @@ var QueryGenerator = {
 
     if (options && options.updateOnDuplicate) {
       onDuplicateKeyUpdate += ' ON DUPLICATE KEY UPDATE ' + options.updateOnDuplicate.map(function(attr) {
-        var key = this.quoteIdentifier(attr);
+        var field = rawAttributes && rawAttributes[attr] && rawAttributes[attr].field || attr;
```

Comment:
`rawAttributes` should always be defined, and field will always be set, so you should be able to shorten this clause